### PR TITLE
Load generation App Updates

### PR DESCRIPTION
**Pull request by PaurushGarg**
**Description:** <Describe what has changed.>

Improves OTLP Metric and Traces Load Generation
1. Remove configurable observation interval, flush interval, and datapoints from metrics- as they are unused and not-required in the context of load-generation required for the performance. 
2. Corrects the command line parameters of metric generation - as per the way it is used from load-gen.
3. updates the test-framework to use the `latest` load-gen app.
4. correctly sets up metric and trace provider to produce the right load (metric/traces).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Danielzolty** [comment]
Here are performance results for 100, 1,000, and 5,000 TPS for metric tests:

100:
`{"testcase":"otlp_metric_mock","instanceType":"m5.2xlarge","receivers":["otlp"],"processors":["batch"],"exporters":["logging","awsemf"],"dataType":"otlp","dataMode":"metric","dataRate":100,"avgCpu":0.350000239262089,"avgMem":109.9917312,"maxCpu":0.5000363635444206,"maxMem":111.120384,"commitId":"dummy_commit","collectionPeriod":10,"testingAmi":"soaking_linux"}`

1,000:
`{"testcase":"otlp_metric_mock","instanceType":"m5.2xlarge","receivers":["otlp"],"processors":["batch"],"exporters":["logging","awsemf"],"dataType":"otlp","dataMode":"metric","dataRate":1000,"avgCpu":2.7616599662192955,"avgMem":141.03784106666666,"maxCpu":8.99966334149369,"maxMem":154.710016,"commitId":"dummy_commit","collectionPeriod":10,"testingAmi":"soaking_linux"}`

5,000:
`{"testcase":"otlp_metric_mock","instanceType":"m5.2xlarge","receivers":["otlp"],"processors":["batch"],"exporters":["logging","awsemf"],"dataType":"otlp","dataMode":"metric","dataRate":5000,"avgCpu":14.974639151005709,"avgMem":224.80576511999996,"maxCpu":21.39946276862715,"maxMem":254.705664,"commitId":"dummy_commit","collectionPeriod":10,"testingAmi":"soaking_linux"}`

**Review by vasireddy99 — COMMENTED** [review]
Did 1st pass on this PR

File: load-generator/README.md [hunk 1]
```diff
@@ -49,7 +41,7 @@ The following is a list of additional optional command line arguments applicable
 
 ### StatsD Metrics Load Test Sample Command,
 ```
-./gradlew :load-generator:run --args="metric --metricCount=100 --datapointCount=10 --observationInterval=1000 -u=localhost:8125 -d=statsd"
```
Comment (vasireddy99) [inline load-generator/README.md:52]: observationInterval=1000 should we have observation interval configurable.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java [hunk 2]
```diff
@@ -29,127 +29,114 @@
...
-        counters[id] = meter.counterBuilder(API_COUNTER_METRIC + id)
-                .setDescription("API request load sent in bytes")
-                .setUnit("one")
-                .build();
-      }
-    } else {
-      log.error("Metric provider was not found!");
+
+    private void createCounters(Meter meter) {
+        if (meter != null) {
```
Comment (rapphil) [inline load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java]: nit: this could be replaced by an early return.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java [hunk 3]
```diff
@@ -29,127 +29,114 @@
...
+                        .setUnit("ms")
+                        .ofLongs().buildWithCallback(measurement ->
+                                {
+                                    Attributes atr = Attributes.of(AttributeKey.stringKey(DIMENSION_API_NAME), API_NAME,
+                                            AttributeKey.stringKey(DIMENSION_STATUS_CODE), "200");
+                                    measurement.record(gaugeValue, atr);
+                                }
+                        );
+            }
+        } else {
```
Comment (rapphil) [inline load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java]: same here.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/MetricEmitter.java [hunk 4]
```diff
@@ -32,8 +31,8 @@ public abstract class MetricEmitter implements Emitter {
 
   @Override
   public void start(Runnable emitter) {
-    log.info("Generating metrics at a rate of(ms) : {}" , this.param.getObservationInterval());
-    scheduler.scheduleAtFixedRate(emitter, 0, this.param.getObservationInterval(), TimeUnit.MILLISECONDS);
+    log.info("Generating metrics at a rate of(ms) : {}" , FLUSH_INTERVAL);
+    scheduler.scheduleAtFixedRate(emitter, 0, FLUSH_INTERVAL, TimeUnit.MILLISECONDS);
```
Comment (rapphil) [inline load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/MetricEmitter.java:35]: this simplifies things a bit. Should we add an warning/raise an exception in case the emitter function is not able to complete in less than 1s? Otherwise the ingestion rate will not be correct right?

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/TraceEmitter.java [hunk 5]
```diff
@@ -24,7 +24,7 @@ public abstract class TraceEmitter implements Emitter {
 
   @Override
   public void start(Runnable emitter) {
-    scheduler.scheduleAtFixedRate(emitter, TimeUnit.SECONDS.toMillis(5000),
+    scheduler.scheduleAtFixedRate(emitter, 0,
         TimeUnit.SECONDS.toNanos(1) / this.param.getRate(), TimeUnit.NANOSECONDS);
```
Comment (rapphil) [inline load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/TraceEmitter.java:28]: I think you need to fix this as well no? this needs to be scheduled every second and emitter should emitt the rate number of spans?

**Comment by bryan-aguilar** [comment]
> Improves OTLP Metric Load Generation

How? By doing what? What changes are being made in this PR?

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/Emitter.java [hunk 6]
```diff
@@ -21,6 +21,9 @@
 public interface Emitter {
 
   int NUM_THREADS = 5;
+
+  final long FLUSH_INTERVAL = 1000;
```
Comment (bryan-aguilar) [inline load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/Emitter.java:25]: Why is this no longer configurable?

File: terraform/testcases/prometheus_mock/otconfig.tpl [hunk 7]
```diff
@@ -5,7 +5,7 @@ receivers:
   prometheus:
     config:
       global:
-        scrape_interval: 15s
+        scrape_interval: 1s
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_mock/otconfig.tpl]: Doesn't changing the scrape interval affect performance? Why is this change necessary?

File: load-generator/README.md (hunk 1 above)
Comment (PaurushGarg) [reply load-generator/README.md:52]: I think as sample app, providing configurable observationInterval is good, but in load generator context and how it is used through test-framework; I think it would be unused functionality and could create problem as well.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java (hunk 2 above)
Comment (PaurushGarg) [reply load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java]: Updated.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java (hunk 3 above)
Comment (PaurushGarg) [reply load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java]: Updated.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/TraceEmitter.java (hunk 5 above)
Comment (PaurushGarg) [reply load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/TraceEmitter.java:28]: Raphael, in my opinion, this is already fixed. works little differently than metrics. nextdatapoint is called no of rate times, which produces that many traces in a second, which gets batched and exported at 1s frequency.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/Emitter.java (hunk 6 above)
Comment (PaurushGarg) [reply load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/Emitter.java:25]: I think, for load gen its not required to be configured, as we have TPS (100, 1000, 5000) batched at 1s. Adding flush_interval just adds unused complexity. 
https://github.com/aws-observability/aws-otel-test-framework/pull/1509#discussion_r1416328784

File: terraform/testcases/prometheus_mock/otconfig.tpl (hunk 7 above)
Comment (PaurushGarg) [reply terraform/testcases/prometheus_mock/otconfig.tpl]: yes it should affect performance by factor of 15. This is to match the metric load as mentioned in the performance report.

Comment (PaurushGarg) [reply terraform/testcases/prometheus_mock/otconfig.tpl]: Reverted. Will pull into separate PR as discussed.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/MetricEmitter.java (hunk 4 above)
Comment (PaurushGarg) [reply load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/MetricEmitter.java:35]: I am unsure how to implement this. `scheduleAtFixedRate` only throws these three exceptions:

> RejectedExecutionException – if the task cannot be scheduled for execution
> NullPointerException – if command or unit is null
> IllegalArgumentException – if period less than or equal to zero

**Comment by PaurushGarg** [comment]
> > Improves OTLP Metric Load Generation
> 
> How? By doing what? What changes are being made in this PR?

Updated PR body.

**Review by bryan-aguilar — APPROVED** [review]
